### ISSUE TITLE
Ranking system for MLR-Algorithms or speed bars.

### ISSUE THREAD

**Issue by IcedragonP** [berndbischl/mlr#286]
Maybe this idea is a bit stupid.

But imagine you are a newbie to Machine Learning and don't know much about the different algorithms and you want to choose only some of the implemented algorithms (e.g. because your data is too big to run all algorithms quickly). 

You could search in google for some comparative papers. These are always dependent on the 
data that was used. 

My idea: 
Make a rating system like in Amazon for the different mlr algorithms. 
Every user can vote once (1-5 stars) for every algorithm, depending on his experiences with 
the algorithms. 
Results could be posted here for example:
http://berndbischl.github.io/mlr/tutorial/html/integrated_learners/index.html

Maybe this is also hard to implement and would need a real website...

Another idea would be a speed bar for each algorithm, so you could see which algorithms are faster (maybe one for variables and one for observations).

**Comment by larskotthoff** [comment]
One problem with this is that how an algorithm performs really depends on the data (and the parameters of the algorithm). That is, if algorithm A performs well on data set X, it doesn't mean that it will perform well on data set Y.

There is already kind of such a thing (http://openml.org/), where you can see the performance of algorithms on data sets. There's no human bias in the evaluation, as there would be in a rating system.

Regarding the speed bar: speed is almost always a secondary consideration, usually the achieved performance is the most important thing. That is, we don't want an algorithm that quickly does something badly, but rather one that does something well.

**Comment by berndbischl** [comment]
Lars' comments are correct, but I fully understand what the OP wants and I also think it can be done, it is simply a lot of work. Although maybe slightly different than the OP described. It is very certainly not a stupid question.

Wrt predictive performance: We could run a pretty exhaustive benchmark of all mlr algos on openml. Aggregared results could be featured on the mlr website. Maybe even including tuned algos. I am pretty sure this could be informative for beginners.

I also completely agree that timing info is not thaz secondary and very relevant. Its display is going to be more complex than a simple speed bar. But it could even be modelled more easily than normal performance by using simple data features. I am currently trying to convince a student to do a first try in his thesis about this. And I wanted to do this for a longer time now as the resulting data could be exploited in many ways IMHO.

That said, I cannot do this now and the tracker is very crowded. Hence I will close this as I will surely not forget this.

If anyone wants to put real work into this, email me and or reopen here.

**Comment by larskotthoff** [comment]
So, for the record, what would be your proposed implementation? I think I don't understand what it would be given the comments so far.

**Comment by berndbischl** [comment]
What exactly do you mean?

**Comment by berndbischl** [comment]
Ok I talked to Lars, and we agreed to briefly document what I meant here.

I would not really implement a "user-based scoring" system (like the OP hinted at), where people "star" their favorite method. What I could be interested in is a continuous benchmarking of predictive performance and speed behavior of all mlr learners on a large portion of OpenML. 
This would generate a wealth of data, that we could exploit in various fashions:

a) Make it available to everybody for own analysis.
 (By uploading to OpenML this would be kinda ensured automatically, but we could also offer the data in a nicer data.frame on the webpage)

b) Extract a "well-performing" subset of learners that work well on many data sets. 
I tried to do something very similar here:
http://link.springer.com/article/10.1007%2Fs00180-013-0420-y

c) Give feedback to users about the potential runtime behavior of algorithms. 

So, closing for now. Still very interested in this topic. Get back to me if you have more time on your hands than I currently have ;-)
